test/NuGet.Core.Tests/NuGet.PackageManagement.Test/BuildIntegratedTests.cs

```diff
@@ -1848,14 +1851,2073 @@ public async Task TestPacMan_BuildIntegrated_PreviewUpdatesAsync_WithStrictVersi
+                    var packageA_Version100 = new SimpleTestPackageContext("packageA", "1.0.0");
+
+                    var packageA = packageA_Version100.Identity;
+
+                    await SimpleTestPackageUtility.CreateFolderFeedV3Async(
+                        pathContext.PackageSource,
+                        PackageSaveMode.Defaultv3,
+                        packageA_Version100
+                        );
+
+                    sources.Add(new PackageSource(pathContext.PackageSource));
+
+                    var sourceRepositoryProvider = TestSourceRepositoryUtility.CreateSourceRepositoryProvider(sources);
+
+                    var deleteOnRestartManager = new TestDeleteOnRestartManager();
+                    var nuGetPackageManager = new NuGetPackageManager(
+                        sourceRepositoryProvider,
+                        NullSettings.Instance,
+                        testSolutionManager,
+                        deleteOnRestartManager);
+
+                    var testNuGetProjectContext = new TestNuGetProjectContext();
+                    var projectTargetFramework = NuGetFramework.Parse("net46");
+
+                    var configs = new List<string>();
+                    var lockFiles = new List<string>();
+                    var buildIntegratedProjects = new List<TestProjectJsonBuildIntegratedNuGetProject>();
+
+                    // Create projects
+                    for (var i = 0; i < 4; i++)
+                    {
+                        var directory = TestDirectory.Create();
+                        projectDirectories.Add(directory);
+
+                        var config = Path.Combine(directory, "project.json");
+
+                        configs.Add(config);
+
+                        GetBasicConfig(config, "net46");
+
+                        var msBuildNuGetProjectSystem = new TestMSBuildNuGetProjectSystem(
+                            projectTargetFramework,
+                            testNuGetProjectContext,
+                            directory,
+                            $"testProjectName{i}");
+
+                        var buildIntegratedProject = new TestProjectJsonBuildIntegratedNuGetProject(config, msBuildNuGetProjectSystem);
+
+                        // We need to treat NU1065 warning as error.
+                        buildIntegratedProject.IsNu1605Error = true;
+
+                        buildIntegratedProjects.Add(buildIntegratedProject);
+
+                        lockFiles.Add(ProjectJsonPathUtilities.GetLockFilePath(config));
+
+                        testSolutionManager.NuGetProjects.Add(buildIntegratedProject);
+
+                        //Let new project pickup my custom package source.
+                        buildIntegratedProject.ProjectLocalSources.AddRange(sources);
+                    }
+
+                    // Link projects
+                    var reference0 = new TestExternalProjectReference(buildIntegratedProjects[0], buildIntegratedProjects[1]);
```

Comment:
consider naming the reference to tell us the FromAndTo projects.

Reply:
I'm removing all unit test from BuildIntegratedTest.cs and ported to NetCorePackageReferenceProjectTests.cs
So we don't need those references anymore.

Reply:
🔔 

You said you'll remove this but it seems like it's survived for now :D